pkg/pillar/dpcreconciler/linux.go

```diff
@@ -965,11 +965,15 @@ func (r *LinuxDpcReconciler) getIntendedPhysicalIfs(dpc types.DevicePortConfig)
 			}
 		case types.L2LinkTypeVLAN:
 			parent := dpc.LookupPortByLogicallabel(port.VLAN.ParentPort)
+			usage := generic.IOUsageVlanParent
+			if parent.IsL3Port {
+				usage = generic.IOUsageVlanParentAndL3Adapter
+			}
```

Comment:
Does this imply that you have slightly redefined what IsL3Port means? 
Or am I not understanding the IOUsage* - Does IOUsageX mean that it is currently used for X, or does it mean that it is available to be used for X?

For example, does IOUsageVlanParentAndL3Adapter mean that it is used as an L3Adapter or that it is capable/available to be used as an L3Adapter?

Reply:
Both `IsL3Port` and the `IOUsage` from DPCReconciler describe the current usage of the adapter (not what it is available for but what it is actually used for in the current DPC).

Reply:
In the case of the code referenced, we see that the port is used with SystemAdapter (i.e. has IP config) and at the same time has VLAN sub-interface, so usage is `IOUsageVlanParentAndL3Adapter` (but note that this IOUsage is an internal type inside DPCReconciler, not part of some pubsub message that would be visible to a user).